### PR TITLE
make global http client worker threads configurable

### PR DESCRIPTION
**Pull request by zhxiaogg**
ref #3301

### CODE REVIEW

**Comment by zhxiaogg** [comment]
The code could be review now. :)

**Comment by drcrallen** [comment]
Thanks @zhxiaogg !   does `io.druid.guice.http.JettyHttpClientModule.HttpClientProvider` need modified as well? (I haven't ever really dug into the http client provider code before)

**Comment by zhxiaogg** [comment]
@drcrallen yes, it also needs to be modified, the default thread pool for jetty http client is configured by maximum 200 threads. The code has been changed.

File: server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java [hunk 1]
```diff
@@ -116,6 +117,9 @@ public HttpClient get()
 
       httpClient.setIdleTimeout(config.getReadTimeout().getMillis());
       httpClient.setMaxConnectionsPerDestination(config.getNumConnections());
+      final QueuedThreadPool queuedThreadPool = new QueuedThreadPool(config.getNumMaxThreads());
```
Comment (drcrallen) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: It looks like this makes non-daemon by default. Suggest setting daemon to true for thread pool

File: server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java [hunk 2]
```diff
@@ -116,6 +117,9 @@ public HttpClient get()
...
+      queuedThreadPool.setName(QueuedThreadPool.class.getSimpleName() + "@" + queuedThreadPool.hashCode());
```
Comment (drcrallen) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: Can this be more descriptive? something like `JettyHttpClientModulePool@HASH` or similar?

File: server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java (hunk 1 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: Unless there's some guarantee that the pool lifecycle properly manages the thread in the overall lifecycle

Comment (zhxiaogg) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: it's managed by jetty life cycle manager.

File: server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java (hunk 2 above)
Comment (zhxiaogg) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: of course, I've update the code.

File: server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java (hunk 1 above)
Comment (drcrallen) [inline server/src/main/java/io/druid/guice/http/JettyHttpClientModule.java]: I see now, cool

**Comment by drcrallen** [comment]
As a side note, I couldn't find where the http configs are documented. There may need to be a ticket to add that somewhere in the docs.

**Comment by zhxiaogg** [comment]
opened a new issue for documentation. cc @drcrallen

**Comment by gianm** [comment]
They're documented in broker.md, properties like "druid.broker.http.readTimeout".

**Comment by gianm** [comment]
Ah, nevermind, that's only the client the broker uses to talk to data nodes. The other client used for other stuff doesn't have its configs documented.

**Pull request merged**